src/panels/alarm-table/action_mgr.js

```diff
@@ -119,10 +127,17 @@ export class ActionMgr {
     this.options.push({
       text: text + this.getSuffix(rows),
       click: () => {
-        // Apply the action to each row in the selection
-        _.each(rows, row => action(row));
+        // Apply the action(s) to each row in the selection, one after another on success
+        async.eachLimit(rows, 1, (row, callback) => action(row, callback), (error) => {
+          if(error){
+            //Log Error and terminate action(s) for other row(s)
+            console.log('Error in clearing: ', error)
+          }
+          else {
+            this.ctrl.refreshDashboard();
```

Comment:
Could it be a problem that the dashboard is getting refreshed multiple times in quick succession?  Is it enough to refresh when the `eachLimit` calls are finished?

Reply:
Because for one action, all the panel data is being fetched after each iteration so just to make less frequent server request we discussed that refreshing in the end should be fine.

Reply:
Right, I just meant that if you are processing a series of actions then it's going to refresh over and over in quick succession.  Is that refreshed UI useful to the user, or should you wait to refresh until all actions have completed?

Reply:
We want all of the actions to be executed in parallel and once they are all complete, the dashboard should be refreshed.

Reply:
AFAICT `async.eachLimit` with a limit of `1` means they won't run in parallel, doesn't it?  Or are the actions already running as promises so they happen at the same time anyway?

Reply:
That's right.. its going one after another. Updated the pull request by making the actions request in parallel and not terminate if error with any of them.
Dashboard refreshing after the last action is executed.
Thanks!